src/Utils/uniqueidx.jl

```diff
@@ -1,48 +1,88 @@
+export uniqueidx, sortunique
 
-export uniqueidx
-
+   
 function uniqueidx( a::Array{Int64,1} )
-    # b = sorted unique(a)
-    # ii, jj are permutations:
-    #    a[ii] = b,  b[jj] = a
-
-    const n = length(a)
-
-    #ii = sortperm(a)
-    #ii = sortperm(a,alg=MergeSort)
-    #b = a[ii]  # sorted array
-    ii,b = sortpermFast(a)
-
-    # jj = invperm(ii)  # inverse permutation
-    jj = Array(Int64,n)
+# b = sorted unique(a)
+# ii, jj are permutations:
+#    a[ii] = b,  b[jj] = a
 
-    nunique = n  # counter for # of unique values
-    p1 = 1  # pointer to the next unique element
-    jj[ii[1]] = 1
+   
+   const n = length(a)
```

Comment:
not arguing about "nice" code at this point. I haven't seen any `const` in jInv yet, so I was wondering if you timed it and found declaring constants made a difference. Is that the case?